### PR TITLE
Issue:4044453:Replace the flask framework with aiohttp in PDR plugin

### PR DESCRIPTION
**Pull request by vg12345**
## What
Replace the flask framework with aiohttp in PDR plugin

## Why ?
[#4044453](https://redmine.mellanox.com/issues/4044453)

## Testing ?
Automatic tests

## Special triggers
_Use the following phrases as comments to trigger different runs_

* ```bot:retest``` rerun Jenkins CI (to rerun GitHub CI, use "Checks" tab on PR page and rerun _all_ jobs)
* ```bot:upgrade``` run additional update tests

### CODE REVIEW

**Comment by vg12345** [comment]
bot:retest

File: plugins/pdr_deterministic_plugin/ufm_sim_web_service/api/base_aiohttp_api.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
+#
+# Copyright © 2013-2022 NVIDIA CORPORATION & AFFILIATES. ALL RIGHTS RESERVED.
```
Comment (ananalaghbar) [inline plugins/pdr_deterministic_plugin/ufm_sim_web_service/api/base_aiohttp_api.py:2]:
```suggestion
# Copyright © 2013-2024 NVIDIA CORPORATION & AFFILIATES. ALL RIGHTS RESERVED.
```

**Review by drorlevy — APPROVED** [review]
looks fine. do we need the base class as wrapper?

**Comment by vg12345** [comment]
> looks fine. do we need the base class as wrapper?

Not necessary but more readable. I decided to define all common, API-specific independent functionality to base class. In addition, we can later move the base class to utils and reuse.

File: plugins/pdr_deterministic_plugin/ufm_sim_web_service/api/base_aiohttp_api.py (hunk 1 above)
Comment (vg12345) [reply plugins/pdr_deterministic_plugin/ufm_sim_web_service/api/base_aiohttp_api.py]: Fixed

**Review by ananalaghbar — APPROVED** [review]

File: plugins/pdr_deterministic_plugin/requirements.txt [hunk 2]
```diff
@@ -1,9 +1,8 @@
-flask<=3.0.3
 numpy<=1.26.4
 pandas<=2.2.2
 pytest<=8.2.0
 requests<=2.31.0
 twisted<=22.1.0
```
[the remark this answers is not in the document]
Comment (vg12345) [reply plugins/pdr_deterministic_plugin/requirements.txt]: Remove references to twisted

**Comment by vg12345** [comment]
bot:retest

**Pull request merged**